src/inmanta/warnings.py

```diff
@@ -140,15 +140,19 @@ def _showwarning(
         :param line: Required for compatibility but will be ignored.
         """
         # implementation based on warnings._showwarnmsg_impl and logging._showwarning
-        text: str = warnings.formatwarning(
-            # ignore type check because warnings.formatwarning accepts Warning instance but it's type definition doesn't
-            message,  # type: ignore
-            category,
-            filename,
-            lineno,
-            line,
-        )
-        logger: logging.Logger = logging.getLogger("py.warnings")
+        if filename == __file__:
+            text: str = "%s: %s" % (category.__name__, message)
+            logger: logging.Logger = logging.getLogger("inmanta.warnings")
```

Comment:
* I would also add some documentation here regarding why we use a different logger for warnings that we generate ourselves.
* We also need to update the documentation of the `inmanta.warnings.warn()` method to indicate that this method should be used to log warnings instead of the `warnings.warn()` method. Currently is still mentions this method is only used by the tests, which is no longer true.
* We also need to update all calls to `warnings.warn()` and change them to use the `inmanta.warnings.warn()` method.

Reply:
* We also need to add the `@stable_api` annotation to the `inmanta.warnings.warn()` method, because this method might be used by handler code as well.
* We also need to add a test to `test_import_entry_points.py`.

Reply:
Let's wait with the implementation of these changes. I would like to discuss this with the team first.